File: tests/src/main/java/com/github/klee0kai/test/net/StoneApi.java
package com.github.klee0kai.test.net;

import com.github.klee0kai.test.qualifiers.di.qualifiers.ProductType;

import java.util.UUID;

public class StoneApi {

    public String apiUrl = null;

    public String token = null;

    public UUID uuid = UUID.randomUUID();

    public static StoneApi create(ProductType productType) {
        StoneApi stoneApi = new StoneApi();
        if (productType != null) switch (productType) {
            case DEBUG -> {
                stoneApi.apiUrl = "https://debug.org";
            }
            case DEMO -> {
                stoneApi.apiUrl = "https://demo.org";
            }
            case RELEASE -> {
                stoneApi.apiUrl = "https://release.org";
            }
        }
        return stoneApi;

    }


    private StoneApi() {

    }

}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/net/StoneApi.kt
package com.github.klee0kai.test.net

import com.github.klee0kai.test.qualifiers.di.qualifiers.ProductType
import java.util.*

class StoneApi private constructor() {
    var apiUrl: String? = null
    var token: String? = null
    var uuid = UUID.randomUUID()

    companion object {
        fun create(productType: ProductType?): StoneApi {
            val stoneApi = StoneApi()
            if (productType != null) when (productType) {
                ProductType.DEBUG -> stoneApi.apiUrl = "https://debug.org"
                ProductType.DEMO -> stoneApi.apiUrl = "https://demo.org"
                ProductType.RELEASE -> stoneApi.apiUrl = "https://release.org"
            }
            return stoneApi
        }
    }
}